src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/DiagnosticsExtensionTests.ps1

```diff
@@ -183,4 +183,87 @@ function Test-DiagnosticsExtensionSupportJsonConfig
         # Cleanup
         Clean-ResourceGroup $rgname
     }
-}
\ No newline at end of file
+}
+
+<#
+.SYNOPSIS
+Test the basic usage of the Add/Remove VMSS diagnostics extension command
+#>
+function Test-VmssDiagnosticsExtension
+{
+    $rgname = Get-ComputeTestResourceName
+    $loc = Get-ComputeVmssLocation
+
+    try
+    {
+        # Common
+        New-AzureRMResourceGroup -Name $rgname -Location $loc -Force;
+
+        # Create VMSS
+
+        # SRP
+        $stoname = 'sto' + $rgname;
+        $stotype = 'Standard_GRS';
+        New-AzureRMStorageAccount -ResourceGroupName $rgname -Name $stoname -Location $loc -Type $stotype;
+        $stoaccount = Get-AzureRMStorageAccount -ResourceGroupName $rgname -Name $stoname;
+
+        # NRP
+        $subnet = New-AzureRMVirtualNetworkSubnetConfig -Name ('subnet' + $rgname) -AddressPrefix "10.0.0.0/24";
+        $vnet = New-AzureRMVirtualNetwork -Force -Name ('vnet' + $rgname) -ResourceGroupName $rgname -Location $loc -AddressPrefix "10.0.0.0/16" -Subnet $subnet;
+        $vnet = Get-AzureRMVirtualNetwork -Name ('vnet' + $rgname) -ResourceGroupName $rgname;
+        $subnetId = $vnet.Subnets[0].Id;
+
+        # New VMSS Parameters
+        $vmssName = 'vmss' + $rgname;
+        $vmssType = 'Microsoft.Compute/virtualMachineScaleSets';
+
+        $adminUsername = 'Foo12';
+        $adminPassword = "BaR@123" + $rgname;
+
+        $imgRef = Get-DefaultCRPImage -loc $loc;
+        $vhdContainer = "https://" + $stoname + ".blob.core.windows.net/" + $vmssName;
+
+        $extname = 'diagextest';
+        $diagExtPublisher = 'Microsoft.Azure.Diagnostics';
+        $diagExtType = 'IaaSDiagnostics';
+
+        # This storage name will be used in command line directly when set diagnostics extension
+        $storagename = 'definedinconfigstorage'
+        $storagetype = 'Standard_GRS'
+        New-AzureRmStorageAccount -ResourceGroupName $rgname -Name $storagename -Location $loc -Type $storagetype
+
+        $ipCfg = New-AzureRmVmssIPConfig -Name 'test' -SubnetId $subnetId;
+        $vmss = New-AzureRmVmssConfig -Location $loc -SkuCapacity 2 -SkuName 'Standard_A0' -UpgradePolicyMode 'automatic' -NetworkInterfaceConfiguration $netCfg `
+            | Add-AzureRmVmssNetworkInterfaceConfiguration -Name 'test' -Primary $true -IPConfiguration $ipCfg `
+            | Set-AzureRmVmssOSProfile -ComputerNamePrefix 'test' -AdminUsername $adminUsername -AdminPassword $adminPassword `
+            | Set-AzureRmVmssStorageProfile -Name 'test' -OsDiskCreateOption 'FromImage' -OsDiskCaching 'None' `
+            -ImageReferenceOffer $imgRef.Offer -ImageReferenceSku $imgRef.Skus -ImageReferenceVersion $imgRef.Version `
+            -ImageReferencePublisher $imgRef.PublisherName -VhdContainer $vhdContainer `
+            | Add-AzureRmVmssDiagnosticsExtension -Name $extname -SettingFilePath "$TestOutputRoot\ConfigFiles\DiagnosticsExtensionPublicConfig.json" `
```

Comment:
Any reason, no tests have not been added for covering rest of the parameters for AddAzureRmVmssdianosticsExtension (e.g. VirtualMachineScaleSet, PrtectedSettingFilePath, TypeHandlerVersion)
If you are planning to add those test later, can you create an issue and link to this PR or if you think those tests are not needed, please mention in the PR.